### DISCUSSION TITLE
Connecting an SPI LCD with ILI9341 and XPT2046

### DISCUSSION THREAD

**Discussion by BitLoose** [df8oe/UHSDR#1931] [Q&A] [answered]
Trying to hook up an SPI LCD, but looking at the latest schematic there is no connection for the D/C pin.
The board has an ILI9341 controller, and the standard XPT2046 touch controller, but the McHF SPI connection uses pins that were dedicated to the touch interface, and there is no data as to how to connect it.
If you know a solution please post, else I guess I will have to code through the code and reassign the I/O.
Is Cris still working on the code?  I´m hoping to get familiar with it so that I can experiment with a few ideas I have, its a great project, and I´m looking forward to learning the code. Thanks

**Comment by db4ple** [comment, marked as answer]
> Trying to hook up an SPI LCD, but looking at the latest schematic there is no connection for the D/C pin. The board has an ILI9341 controller, and the standard XPT2046 touch controller, but the McHF SPI connection uses pins that were dedicated to the touch interface, and there is no data as to how to connect it. 

I don't understand your statement. The SPI displays and touch controller share the same SPI bus (with dedicated CS pins for each device). 
So for the display you would the same SPI pins as shown for the touch in the schematics. With the exception of the LCD_CS pin of course, which is the same for parallel and SPI displays, so it is also visible in the schematics.
 
I don't know the ILI9341 controller, but if the D/C pin is used to tell the controller if data or commands are being sent, then you can use the LCD_RS pin (D11), which does the same thing for the ILI9486  controller. 

See
https://github.com/df8oe/UHSDR/blob/dcbebe5f79f28066a13e6328446b7c0bc6009c27/mchf-eclipse/drivers/ui/lcd/ui_lcd_hy28.c#L702-L710 
 
>If you know a solution please post, else I guess I will have to code through the code and reassign the I/O. Is Cris still working on the code? I´m hoping to get familiar with it so that I can experiment with a few ideas I have, its a great project, and I´m looking forward to learning the code. Thanks

AFAIK Chris is not working on the code since a long time.

**Reply by BitLoose** [reply]
Thanks, much appreciated, I'll take a look at the CS lines again, I saw "CS a" but no "CS T/touch", so makes sense they share the same SPI bus.  It's really a help pointing out the relevant code defines, thanks again.
(Edit)   Do you know why SPI DMA is only supported with STM32H7?  32F407 also has SPI DMA support
#ifndef STM32H7
  // FIXME: H7 Port, re-enable DMA once SPI display is working
  #define USE_SPI_DMA
#endif

**Comment by sp9bsl** [comment]
@BitLoose in case you have older pcb for UI, you can find suitable info for mods [on this page](https://github.com/df8oe/UHSDR/wiki/Recommended-Modifications#ui-04-hy28b)

This also might be helpfull understanding how things works since years.
Starting with pcb version 6.3, Chris removed parallel connection to LCD (latest schematics on his web page).

**Reply by BitLoose** [reply]
That is super useful, thanks a lot!  I see how the touch IRQ line connects to the DMA channel line now, great!!
Are you working on the code?

**Reply by sp9bsl** [reply]
Trying to back to business... Anyway we're working ratcher with newer hardware based on FPGA than old mixer solution. As Danilo mentioned we are using larger displays with twice more points compared to original design. All ideas are described in our forum [here](https://www.amateurfunk-sulingen.de/forum/index.php). Look into OVI40 / SParkle threads.

**Reply by BitLoose** [reply]
Okay yes, I see the Wolf TX mentioned which has a large FPGA chip.
I tried to register on the forum but it threw this error: 2: file(): SSL operation failed with code 1. OpenSSL Error messages: error:0A000126:SSL routines::unexpected eof while reading
(/var/www/forum/Sources/Register.php ln 325)
